### PR TITLE
#102 Improved user experience

### PR DESCRIPTION
**Pull request by Chopikov**
- Fixed bug with image before cropping when the screen is rotated

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 1]
```diff
@@ -219,17 +229,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
                     .getParcelable(CROP_LOADED_IMAGE);
-            try {
-                _bmpCurrentImage
-                        = getBitmap(this.getContentResolver(),
-                                _uriCurrentImage);
-            } catch (IOException e) {
-                e.printStackTrace();
+
+            _isCropped = savedInstanceState.getBoolean(BMP_IS_CROPPED_TAG);
+            if(_isCropped) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: It needs space

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 2]
```diff
@@ -219,17 +229,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                try {
+                    _bmpCurrentImage
+                            = getBitmap(this.getContentResolver(),
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: `=` needs to be on the previous line.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 3]
```diff
@@ -219,17 +229,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                            _uriCurrentImage);
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: It needs eight spaces.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 4]
```diff
@@ -302,6 +319,7 @@ public void onSaveInstanceState(Bundle outState) {
         _uriCurrentImage = writeTempStateStoreBitmap(this,
                 _bmpCurrentImage, _uriCurrentImage);
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: It needs one more line.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 5]
```diff
@@ -302,6 +319,7 @@ public void onSaveInstanceState(Bundle outState) {
...
         outState.putParcelable(CROP_LOADED_IMAGE, _uriCurrentImage);
+        outState.putBoolean(BMP_IS_CROPPED_TAG, _isCropped);
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: Use `putByte`.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 6]
```diff
@@ -219,17 +229,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
             _uriCurrentImage = (Uri)savedInstanceState
                     .getParcelable(CROP_LOADED_IMAGE);
-            try {
-                _bmpCurrentImage
-                        = getBitmap(this.getContentResolver(),
-                                _uriCurrentImage);
-            } catch (IOException e) {
-                e.printStackTrace();
+
+            _isCropped = savedInstanceState.getBoolean(BMP_IS_CROPPED_TAG);
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: Use `getByte`.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java (hunk 2 above)
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: You can no use `this`.

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 7]
```diff
@@ -219,17 +229,23 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            } catch (IOException e) {
-                e.printStackTrace();
+            _isCropped = savedInstanceState
+                    .getByte(BMP_IS_CROPPED_TAG, (byte) 0) != 0;
+            if (_isCropped) {
+                try {
+                    _bmpCurrentImage = getBitmap(getContentResolver(),
+                            _uriCurrentImage);
+                } catch (IOException e) {
+                    e.printStackTrace();
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java:243]: Call `setImageFromPath(_pathImage);` here.

**Review by maik-nack — APPROVED** [review]
Good job!

**Review by MaximSmolskiy — APPROVED** [review]
Looks great!

**Pull request merged**